src/Parser/ASTConverter.cs

```diff
@@ -940,6 +940,24 @@ void VisitDeclaration(Declaration decl, AST.Declaration _decl)
             _decl.IsIncomplete = decl.IsIncomplete;
             _decl.IsDependent = decl.IsDependent;
             _decl.DefinitionOrder = decl.DefinitionOrder;
+            for (uint i = 0; i < decl.RedeclarationsCount; i++)
+            {
+                var redecl = decl.GetRedeclarations(i);
+                if (Declarations.ContainsKey(redecl.OriginalPtr))
```

Comment:
Do not do this yourself but rather call redecl.Visit. It checks the dictionary for you.